docs/conf.py

```diff
@@ -14,6 +22,8 @@
 project = "scikit-build-core"
 copyright = "2022, The Scikit-Build admins"
 author = "Henry Schreiner"
+version = importlib_metadata.version("scikit-build-core")
```

Comment:
AFAIK this is the recommended way to get the version, and you are supposed to install the package to build the docs. Is that feasible, @kloczek? Otherwise, would putting an environment variable override help?